src/main/java/bio/terra/service/load/flight/LoadUnlockStep.java

```diff
@@ -1,33 +1,78 @@
 package bio.terra.service.load.flight;
 
+import bio.terra.service.dataset.flight.ingest.IngestUtils;
 import bio.terra.service.job.DefaultUndoStep;
 import bio.terra.service.job.JobMapKeys;
 import bio.terra.service.load.LoadLockKey;
 import bio.terra.service.load.LoadService;
+import bio.terra.service.load.exception.LoadLockFailureException;
 import bio.terra.stairway.FlightContext;
 import bio.terra.stairway.StepResult;
+import com.google.common.annotations.VisibleForTesting;
+import java.util.Optional;
 
 public class LoadUnlockStep extends DefaultUndoStep {
   private final LoadService loadService;
+  private final String userSuppliedLockName;
 
   /**
-   * This step is meant to be shared by dataset and filesystem flights for unlocking the load tag.
+   * This step is meant to be shared by dataset and filesystem flights for removing {@code
+   * lockName}'s lock held on the load tag in the targeted dataset.
    *
    * <p>It expects the following to be available in the flight context:
    *
    * <ul>
-   *   <li>{@link LoadMapKeys#LOAD_TAG} in the input parameters or working map
+   *   <li>(Optional) {@link LoadMapKeys#LOAD_TAG} in the input parameters or working map (if
+   *       unavailable, any lock associated with the dataset and {@code lockName} will be cleared)
```

Comment:
I think conditional behavior like this makes it harder to maintain steps. Would it be hard to have owning flight always provide the lock name in `LoadMapKeys.LOAD_TAG`? It looks like the only case affected by that change would be `DatasetUnlockFlight`, which is only called from a single place.